aioesphomeapi/connection.py

```diff
@@ -116,6 +118,12 @@ async def _cleanup(self) -> None:
             await self._frame_helper.close()
             self._frame_helper = None
 
+        if self._process_task is not None:
+            self._process_task.cancel()
+            with suppress(asyncio.CancelledError):
+                await self._process_task
+                self._process_task = None
```

Comment:
```suggestion
            with suppress(asyncio.CancelledError):
                await self._process_task
            self._process_task = None
```

The await will likely raise but I think we still want to set it to `None`

Reply:
Good point